src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertySymbolReferenceFinder.cs

```diff
@@ -114,8 +120,14 @@ private static bool IsForEachProperty(IPropertySymbol symbol)
                 ? await FindIndexerReferencesAsync(symbol, document, semanticModel, options, cancellationToken).ConfigureAwait(false)
                 : ImmutableArray<FinderLocation>.Empty;
 
+            var lengthOrCountReferences = IsLengthProperty(symbol) || IsCountProperty(symbol)
+                ? await FindReferencesInElementAccessExpressionsAsync(
+                    symbol, document, semanticModel, isFindMethod: false, cancellationToken).ConfigureAwait(false)
```

Comment:
in general, instead of the boolea, you would pass a lambda in doing the distinctive work ehre.